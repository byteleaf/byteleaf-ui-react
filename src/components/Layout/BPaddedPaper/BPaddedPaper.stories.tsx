import React from 'react';

import Typography from '@material-ui/core/Typography';
import { color } from '@storybook/addon-knobs';

import { BPaddedPaper } from './BPaddedPaper';

export default {
  title: 'Layout|BPaddedPaper',
};

export const BPaddedPaperStory = () => {
  const label = 'Background Color';
  const defaultValue = '#fff';
  const backgroundColor = color(label, defaultValue);

  return (
    <BPaddedPaper backgroundColor={backgroundColor}>
      <Typography>This is a Paper with a small padding and a gray background.</Typography>
    </BPaddedPaper>
  );
};

BPaddedPaperStory.story = {
  name: 'Default',
  parameters: {
    notes:
      "A Paper with a small padding on all sides, so that the content doesn't stick to the paper's border. Gray background",
  },
};
