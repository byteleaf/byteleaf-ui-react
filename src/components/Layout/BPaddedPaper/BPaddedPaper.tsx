/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { Paper, styled, Theme } from '@material-ui/core';

interface PaddedPaperStyleProps {
  backgroundColor?: string;
}

interface PaddedPaperProps {
  backgroundColor?: string;
  children: JSX.Element | JSX.Element[];
}

export const BPaddedPaper = styled(
  ({
    backgroundColor = '#ffffff',
    ...other
  }: PaddedPaperStyleProps & Omit<PaddedPaperProps, keyof PaddedPaperStyleProps>) => <Paper {...other} />,
)(({ theme, backgroundColor }: { theme: Theme } & PaddedPaperStyleProps) => ({
  backgroundColor,
  padding: theme.spacing(4, 4),
}));
