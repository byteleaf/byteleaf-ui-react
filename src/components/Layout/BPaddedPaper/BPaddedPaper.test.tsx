import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BPaddedPaper } from './BPaddedPaper';

describe('BPaddedPaper', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BPaddedPaper>
        <div>Layout</div>
      </BPaddedPaper>,
    );

    expect(container).toMatchSnapshot();
  });

  it('has a gray background', async () => {
    const { container } = render(
      <BPaddedPaper backgroundColor="#ddd">
        <div>Layout</div>
      </BPaddedPaper>,
    );

    expect(container).toMatchSnapshot();
  });
});
