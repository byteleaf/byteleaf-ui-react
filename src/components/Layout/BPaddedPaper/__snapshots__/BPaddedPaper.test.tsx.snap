// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BPaddedPaper has a gray background 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 Component-root-30 Component-root-31 MuiPaper-rounded"
  >
    <div>
      Layout
    </div>
  </div>
</div>
`;

exports[`BPaddedPaper renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 Component-root-1 Component-root-2 MuiPaper-rounded"
  >
    <div>
      Layout
    </div>
  </div>
</div>
`;
